### ISSUE TITLE
[Feature Request]

### ISSUE THREAD

**Issue by Devoxin** [DV8FromTheWorld/JDA#733]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

Hey there! Before you report a bug or suggest a new feature,
  please make sure to follow these steps first!
  
- [x] I have checked for similar issues.
- [x] I have updated to the [latest JDA version][download].
- [x] I have checked the branches or the maintainers' PRs for upcoming features/bug fixes.

> This is not the place to learn Java. Please refer to [StackOverflow][stack overflow]
  for your general programming questions.

### Issue Type

- [ ] Bug Report
- [x] Feature Request

> For questions join the official JDA discord server: https://discord.gg/0hMr4ce0tIl3SLv5

### Description

Discord's API supports retrieving a single ban entry by specifying the user ID as the last parameter in the URL (`/guilds/{guild.id}/bans/{user.id}`), however JDA does not support this functionality. It would be great if this could be added to JDA :)

**Comment by Chiearf-Chaotic** [comment]
As i saw in [Guild.java](https://github.com/DV8FromTheWorld/JDA/blob/release/3.7.1/src/main/java/net/dv8tion/jda/core/entities/Guild.java) and [GuildImpl.java](https://github.com/DV8FromTheWorld/JDA/blob/cb3ae62c31017a1a1b262382b07bb7e6e85c9284/src/main/java/net/dv8tion/jda/core/entities/impl/GuildImpl.java), There is an existent method for retrieval of bans.

But since this is a feature request, i think the best way to get around this is to create a cache method for the functionality that was requested by @Devoxin .This is because getXCache methods, it usually involves the use of SnowflakeCacheView / CacheView which can be useful in extracting the specific ban of a user via user id. If i remember correctly, it is read-only which sounds conventional enough for this type of feature.

**Comment by Devoxin** [comment]
If what you're referring to is what I think it is, that retrieves the **entire** list of guild bans. It would most likely be faster and far more efficient to allow retrieving a single ban if you already have the user ID.

**Comment by Chiearf-Chaotic** [comment]
![image](https://user-images.githubusercontent.com/34756342/43666183-a8ee063a-976a-11e8-8dc6-3057251feef9.png)
I looked through Discord API docs and found that the url you mentioned about in your description only is matched to the 2 descriptions in the image. About getting the bans specifically of a user only was not stated in the docs.
![image](https://user-images.githubusercontent.com/34756342/43666280-19d67850-976b-11e8-8dbe-16b4e415dc4f.png)

**Comment by Devoxin** [comment]
![req](https://user-images.githubusercontent.com/15076404/43666344-617462e4-976b-11e8-948b-83de887880c8.png)

**Comment by Devoxin** [comment]
Also see: https://github.com/discordapp/discord-api-docs/issues/575

**Comment by Sanduhr32** [comment]
Nice research you have done there! Hmm i am going to try if and how well i can implement it and probably am going to pr it if it works like a charm. Would `Guild#retrieveSingleBan()` a good name for the method?

**Comment by Chiearf-Chaotic** [comment]
yes

**Comment by Devoxin** [comment]
I was thinking `getBan` for consistency with `getBans`

**Comment by Chiearf-Chaotic** [comment]
@Devoxin Out of interest, what is the url of the website you were testing the entry from ?

**Comment by Devoxin** [comment]
I was using a desktop application called Postman

**Comment by Sanduhr32** [comment]
Yeah you are right it should be get what about `#getSingleBan()` to make it clear so you can decide between the list and the single ban?

**Comment by Devoxin** [comment]
Sounds good

**Comment by averen** [comment]
I noticed in your URL that it uses v7. Just double checked to verify it works on v6. Good to go.

**Comment by Sanduhr32** [comment]
Hit me up on Discord if you wanna know progress / testing results! Thanks for the v6 check! :D

**Comment by kantenkugel** [comment]
would prefer a getBanById or or similar, as pretty much every (if not every) method using the id for lookup uses the ById suffix

**Comment by natanbc** [comment]
maybe `retrieveBanById`, just like `JDA#retrieveUserById`?

**Comment by kantenkugel** [comment]
not sure... get would be more consistent with getBans()... we only have retrieveUser cuz its different from getUser, which uses the cache compared to a REST call...

**Comment by Sanduhr32** [comment]
I will rename them now from `#getSingleBan()` to `#getBanById()` and `#getBan(User)`.
